packages/alfa-rules/src/sia-r81/rule.ts

```diff
@@ -0,0 +1,157 @@
+import { Rule } from "@siteimprove/alfa-act";
+import { Role } from "@siteimprove/alfa-aria";
+import { Element, Namespace } from "@siteimprove/alfa-dom";
+import { Iterable } from "@siteimprove/alfa-iterable";
+import { List } from "@siteimprove/alfa-list";
+import { Map } from "@siteimprove/alfa-map";
+import { Option } from "@siteimprove/alfa-option";
+import { Predicate } from "@siteimprove/alfa-predicate";
+import { Err, Ok } from "@siteimprove/alfa-result";
+import { Set } from "@siteimprove/alfa-set";
+import { Page } from "@siteimprove/alfa-web";
+
+import * as aria from "@siteimprove/alfa-aria";
+import * as dom from "@siteimprove/alfa-dom";
+
+import { expectation } from "../common/expectation";
+
+import { hasAccessibleName } from "../common/predicate/has-accessible-name";
+import { hasRole } from "../common/predicate/has-role";
+import { isIgnored } from "../common/predicate/is-ignored";
+
+import { Question } from "../common/question";
+
+const { isElement, hasNamespace, hasId } = Element;
+const { map, flatMap, isEmpty } = Iterable;
+const { and, or, not, equals } = Predicate;
+
+export default Rule.Atomic.of<Page, Iterable<Element>, Question>({
+  uri: "https://siteimprove.github.io/sanshikan/rules/sia-r81.html",
+  evaluate({ device, document }) {
+    return {
+      applicability() {
+        const elements = document
+          .descendants({ flattened: true, nested: true })
+          .filter(
+            and(
+              isElement,
+              and(
+                hasNamespace(Namespace.HTML, Namespace.SVG),
+                hasRole(
+                  or(Role.hasName("link"), (role) =>
+                    role.inheritsFrom(Role.hasName("link"))
```

Comment:
Here also we might want an "inclusive inherits" helper as it happens quite a few times.